### PR TITLE
[WFCORE-2787] SSL+Elytron slave connecting to master tests

### PR DESCRIPTION
**Pull request by LittleJohnII**
Project Jira: https://issues.jboss.org/browse/WFCORE-2787
Product Jira: https://issues.jboss.org/browse/JBEAP-10826
Planning Jira: https://issues.jboss.org/browse/EAP7-632

Adds two test cases, first for one-way SSL, second for two-way SSL between master and slave.

I run the whole TS before PR, everything looks OK.

### CODE REVIEW

**Comment by luck3y** [comment]
this is ok to test

**Comment by LittleJohnII** [comment]
I don't see the failures locally, but I'll have to investigate since this is clearly something this PR introduced.

**Comment by darranl** [comment]
As the error messages include IPv6 addresses I would probably start looking there first.

**Comment by LittleJohnII** [comment]
I didn't parametrize the master and slave interface IP addresses and the ssl-context wasn't getting used becuse it was bound to an IPv4 address. This should be ready for review.

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java [hunk 1]
```diff
@@ -559,6 +557,7 @@ public boolean areServersStarted() {
 //            serverStatuses.putAll(statuses);
             return true;
         } catch (Exception ignored) {
+            log.severe(ignored.getLocalizedMessage());
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java]: Please remove.

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java [hunk 2]
```diff
@@ -570,8 +569,10 @@ public boolean isHostControllerStarted() {
             address.add("host", configuration.getHostName());
 
             ControlledProcessState.State status = Enum.valueOf(ControlledProcessState.State.class, readAttribute("host-state", address).asString().toUpperCase(Locale.ENGLISH));
+            log.warning("status: " + status);
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java]: Please remove.

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java [hunk 3]
```diff
@@ -570,8 +569,10 @@ public boolean isHostControllerStarted() {
...
             return status == ControlledProcessState.State.RUNNING;
         } catch (Exception ignored) {
+            log.severe(ignored.getLocalizedMessage());
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/domain/management/util/DomainLifecycleUtil.java]: Please remove.

**Comment by LittleJohnII** [comment]
Sorry for the stupid logging leftover, fixed now. And thanks for the check.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java [hunk 4]
```diff
@@ -117,6 +122,23 @@ protected static void checkHostStatusOnMaster(String host) throws Exception {
         Assert.fail("Cannot validate host '" + host + "' is running");
     }
 
+    protected static void checkHostStatusOnMasterOverRemote(String host) throws IOException, InterruptedException {
+        try (ModelControllerClient mcc = ModelControllerClient.Factory.create(
+                InetAddress.getByName(TestSuiteEnvironment.getServerAddress()), 9999, Authentication.getCallbackHandler())) {
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java]: Why create a client for this? DomainTestSupport.getDomainMasterLifecycleUtil().getDomainClient() should provide a client.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java [hunk 5]
```diff
@@ -117,6 +122,23 @@ protected static void checkHostStatusOnMaster(String host) throws Exception {
...
+            ModelNode result = mcc.execute(getHostStateRunningOperation(host));
+
+            final long time = System.currentTimeMillis() + TIMEOUT;
+            do {
+                if (checkResultIsRunning(result)) {
+                    return;
+                }
+                Thread.sleep(100);
+            } while (System.currentTimeMillis() < time);
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java]: L128 needs to be in the do loop.

**Pull request merged**